core/modules/memman/MemFile.cc

```diff
@@ -0,0 +1,179 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "memman/MemFile.h"
+
+// System Headers
+#include <errno.h>
+#include <mutex>
+#include <unordered_map>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+/******************************************************************************/
+/*                  L o c a l   S t a t i c   O b j e c t s                   */
+/******************************************************************************/
+  
+namespace {
+std::mutex                                 cacheMutex;
+std::unordered_map<std::string, MemFile *> fileCache;
+}
```

Comment:
It works and it's legitimate, but it is buried very deep. In the off chance 2 memory managers are created and one of them, let's say B, releases a file that doesn't belongs to A due to a bug, it could be very difficult to notice this underlying connection between the 2 different objects for people not familiar with the code.